### PR TITLE
Fix Perch2 gpu device string and forward() default targets

### PR DESCRIPTION
**Pull request by benmunkres**
This PR fixes both issues mentioned in #50
1) The GPU device string is switched from "gpu" to "cuda" to prevent an invalid torch.device() call, and the perch2 version selection is updated as such - both "cuda" and "xla" are treated as valid GPU devices (since AMD HIP GPUs can be accessed by torch using the CUDA interface) - this could also be restricted to only "cuda"
2) the default targets for the .forward() method have been changed from `targets=("logits", "embeddings", "spatial_embeddings", "spectrograms")`  to `targets=("logit", "embedding", "spatial_embedding", "spectrogram")` to match the keys in the dict output by the TF model (otherwise it returns a dict of empty lists) 

I've tested with `device = "cpu"` and `device = "cuda"` (both supported versions of Perch2)

### CODE REVIEW

**Comment by benmunkres** [comment]
Since it's a (semi-)related issue I also implemented a fix for #45 - Sam's fix didn't work for me (tf automatically selected the GPU anyways), but using the tf.device() context manager for the model loading and for inference, I was able to get tf to use the CPU for inference if it's selected.

**Comment by benmunkres** [comment]
I re-did the fix for the mismatched target name issue - since forward expects the plural names (and logits instead of label), I changed the fix to re-map the keys of model_outputs (using a class-level dict) - I'm guessing this would make it more consistent with other models (not sure what the conventions are) and would maintain better backwards compatibility

**Comment by sammlapp** [comment]
thanks @benmunkres !
